### ISSUE TITLE
Issues with JS code loading behavior.

### ISSUE THREAD

**Issue by MrJoy** [avo-hq/avo#1632]
### Describe the bug

Various things that I'm probably misunderstanding about JS loading in the context of Propshaft generally, and Avo specifically.

### Steps to Reproduce

See: <https://github.com/MrJoy/collection-repro>

Upon loading the root page, you should see an error about rails-ujs already bein started in the browser console.

Note that all logging statements show a `window` object that has a `Stimulus` property, but that `window.Stimulus` is `undefined` from the perspective of the JS code.

### Expected behavior & Actual behavior

If there's a "right" way to be setting things up that I'm missing, I'd like to understand what it is.  If not -- or if I'm more or less doing things "right", I'd like to understand if my workaround (see comments in `controllers/application.js`) is safe/reasonable.  I'd also like to know if this apparent double-initialization of Stimulus is "dangerous" in some way.

### System configuration
**Avo version**: 2.28.0

**Rails version**: 7.0.4.3

**Ruby version**: 3.1.2p20

**License type**:

 - [x] Community
 - [ ] Pro

**Are you using Avo monkey patches or overriding views or view components?**

 - [ ] Yes. If so, please post code samples.
 - [x] No

#### Impact

 - [ ] High impact
 - [ ] Medium impact
 - [x] Low impact

#### Urgency

 - [ ] High urgency
 - [x] Medium urgency
 - [ ] low urgency

Only listing this as "medium" because this feels like the sort of thing where hacking it until it works without understanding the issue might create a hard-to-unwind mess.

**Comment by adrianthedev** [comment]
![](https://media.giphy.com/media/3o7TKFjfeq2yZHloQw/giphy.gif)

I managed to install, run, and reproduce the issue using your steps.
Let me have a look in the next following days and get back to you with a resolution.

**Comment by adrianthedev** [comment]
![](https://media.giphy.com/media/3orieUe6ejxSFxYCXe/giphy.gif)

TL;DR; add `defer: true` to your `javascript_include_tag`.

I found the issue.
Avo's JS content is loaded with `defer: true`, so it doesn't block the page load. In your repo, the `javascript_include_tag` is there without the `defer: true` option so it's evaluated before ours, so before `window.Stimulus`.

I updated the [docs](https://docs.avohq.io/2.0/custom-asset-pipeline.html#sprockets-and-propshaft) too.

<img width="762" alt="CleanShot 2023-03-19 at 17 46 36@2x" src="https://user-images.githubusercontent.com/1334409/226187578-742ba75a-72c9-454f-82d3-f9cca29d5417.png">
